### DISCUSSION TITLE
RP2040 (Pi Pico) homing issue - Z stepper motor doesn't stop despite z limit switch trigger

### DISCUSSION THREAD

**Discussion by Hyakutake12** [Paciente8159/uCNC#528] [Config and issues] [answered]
Hi. On RP2040 Pi Pico board when homing is enabled and run, the Z stepper motor continues to move regardless of hitting the limit switch and triggering it. 

During homing when the limit switch is triggered, DRO on UGS reads out the Z alarm and $P shows input state as..
[SO:1:0]
[SO:2:0]
[SO:3:0]
[SO:9:0]
[SO:10:0]
[SO:11:1]
[SO:17:0]
[SO:18:0]
[SO:19:0]
[SI:100:0]
[SI:101:0]
[SI:102:1]
[STEPS:0,0,-474]
[RUNTIME:2107677]
..but the z stepper doesn't stop.

I checked the same limit switches on Arduino Uno and switches/homing worked fine. How can i get this to work for pi pico board?
Attaching override and config setting files.. thanks

[boardmap_overrides.txt](https://github.com/Paciente8159/uCNC/files/12848923/boardmap_overrides.txt)
[cnc_hal_overrides.txt](https://github.com/Paciente8159/uCNC/files/12848924/cnc_hal_overrides.txt)
[ucnc_build.txt](https://github.com/Paciente8159/uCNC/files/12848925/ucnc_build.txt)

**Comment by Paciente8159** [comment, marked as answer]
Hi there,

Inside cnc_config.h enable option `FORCE_SOFT_POLLING`.
I will add that option to the web builder.

This will make the limits work properly.
I will try to get the limits change ISR to work ASAP.
